### ISSUE TITLE
XmlReportAppenderTrait does not validate the file to import exists

### ISSUE THREAD

**Issue by discordier** [phpcq/plugin-api#8]
We should check for file existence here: https://github.com/phpcq/plugin-api/blob/7570d992a8d49c46658716eb4a82700454532c3e/src/Version10/Util/XmlReportAppenderTrait.php#L213

**Comment by discordier** [comment]
Just experienced it in one of our travis builds:
```
PHP Warning:  DOMDocument::load(): Document is empty in /tmp/phpcq-5f7ad6b4d5b1c/phpunit5f7ad6b4d849b.junit-log.xml, line: 1 in /home/travis/build/phpcq/phpcq/vendor/phpcq/plugin-api/src/Version10/Util/XmlReportAppenderTrait.php on line 213
PHP Stack trace:
PHP   1. {main}() /home/travis/build/phpcq/phpcq/phpcq:0
PHP   2. Symfony\Component\Console\Application->run() /home/travis/build/phpcq/phpcq/phpcq:50
PHP   3. Symfony\Component\Console\Application->doRun() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Application.php:142
PHP   4. Symfony\Component\Console\Application->doRunCommand() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Application.php:266
PHP   5. Phpcq\Runner\Command\RunCommand->run() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Application.php:920
PHP   6. Phpcq\Runner\Command\RunCommand->execute() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Command/Command.php:258
PHP   7. Phpcq\Runner\Command\RunCommand->doExecute() /home/travis/build/phpcq/phpcq/src/Command/AbstractCommand.php:115
PHP   8. Phpcq\Runner\Command\RunCommand->runTasks() /home/travis/build/phpcq/phpcq/src/Command/RunCommand.php:168
PHP   9. Phpcq\Runner\Task\TaskScheduler->run() /home/travis/build/phpcq/phpcq/src/Command/RunCommand.php:187
PHP  10. Phpcq\Runner\Task\TaskScheduler->tick() /home/travis/build/phpcq/phpcq/src/Task/TaskScheduler.php:86
PHP  11. Phpcq\Runner\Task\ParallelizableProcessTask->tick() /home/travis/build/phpcq/phpcq/src/Task/TaskScheduler.php:108
PHP  12. {anonymous-class:/home/travis/build/phpcq/phpcq/.phpcq/plugins/phpunit/plugin.php:69-97}->finish() /home/travis/build/phpcq/phpcq/src/Task/ParallelizableProcessTask.php:134
PHP  13. Phpcq\PluginApi\Version10\Util\JUnitReportAppender::appendFileTo() /home/travis/build/phpcq/phpcq/.phpcq/plugins/phpunit/plugin.php:92
PHP  14. DOMDocument->load() /home/travis/build/phpcq/phpcq/vendor/phpcq/plugin-api/src/Version10/Util/XmlReportAppenderTrait.php:213
Warning: DOMDocument::load(): Document is empty in /tmp/phpcq-5f7ad6b4d5b1c/phpunit5f7ad6b4d849b.junit-log.xml, line: 1 in /home/travis/build/phpcq/phpcq/vendor/phpcq/plugin-api/src/Version10/Util/XmlReportAppenderTrait.php on line 213
Call Stack:
    0.0002     424360   1. {main}() /home/travis/build/phpcq/phpcq/phpcq:0
    0.0157    2117928   2. Symfony\Component\Console\Application->run() /home/travis/build/phpcq/phpcq/phpcq:50
    0.0204    2366768   3. Symfony\Component\Console\Application->doRun() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Application.php:142
    0.0208    2370520   4. Symfony\Component\Console\Application->doRunCommand() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Application.php:266
    0.0208    2370520   5. Phpcq\Runner\Command\RunCommand->run() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Application.php:920
    0.0211    2372240   6. Phpcq\Runner\Command\RunCommand->execute() /home/travis/build/phpcq/phpcq/vendor/symfony/console/Command/Command.php:258
    0.0409    3195328   7. Phpcq\Runner\Command\RunCommand->doExecute() /home/travis/build/phpcq/phpcq/src/Command/AbstractCommand.php:115
    0.0561    4291520   8. Phpcq\Runner\Command\RunCommand->runTasks() /home/travis/build/phpcq/phpcq/src/Command/RunCommand.php:168
    0.0564    4325784   9. Phpcq\Runner\Task\TaskScheduler->run() /home/travis/build/phpcq/phpcq/src/Command/RunCommand.php:187
   49.4568    4656592  10. Phpcq\Runner\Task\TaskScheduler->tick() /home/travis/build/phpcq/phpcq/src/Task/TaskScheduler.php:86
   49.4570    4656704  11. Phpcq\Runner\Task\ParallelizableProcessTask->tick() /home/travis/build/phpcq/phpcq/src/Task/TaskScheduler.php:108
   49.4571    4646048  12. {anonymous-class:/home/travis/build/phpcq/phpcq/.phpcq/plugins/phpunit/plugin.php:69-97}->finish() /home/travis/build/phpcq/phpcq/src/Task/ParallelizableProcessTask.php:134
   49.4576    4664176  13. Phpcq\PluginApi\Version10\Util\JUnitReportAppender::appendFileTo() /home/travis/build/phpcq/phpcq/.phpcq/plugins/phpunit/plugin.php:92
   49.4576    4664288  14. DOMDocument->load() /home/travis/build/phpcq/phpcq/vendor/phpcq/plugin-api/src/Version10/Util/XmlReportAppenderTrait.php:213
```

The only failing tool was phpunit, which somehow died and therefore did not produce any results.
I think we should handle this by throwing an exception in the Appender class and catching that one in the transformer function of the plugin which then will add a `fatal` diagnostic containing the environmental setup / command line arguments to provide hints to the developer.

**Comment by discordier** [comment]
Fixed with f33240c and in the various plugins
